src/engine/ruleset/decoders/windows/windows-security.yml

```diff
@@ -0,0 +1,326 @@
+
+    # Set Trust Direction
+    - winlog.trustDirection: +kvdb_get/win_trust_direction/$~windows.Event.EventData.TdoDirection
+
+    # Set Trust Attributes
+    - winlog.trustAttribute: +kvdb_get/win_trust_attribute/$~windows.Event.EventData.TdoAttributes
+
+    - winlog.logon.id: $~windows.Event.EventData.SubjectLogonId
+
+    - winlog.event_data.PrivilegeList: '+s_to_array/$~windows.Event.EventData.PrivilegeList/ '
+
+    - user.target.name: $~windows.Event.EventData.OldTargetUserName
+    - user.changes.name: $~windows.Event.EventData.NewTargetUserName
+    - related.user: +a_append/$~windows.Event.EventData.NewTargetUserName
+    - related.user: +a_append/$~windows.Event.EventData.OldTargetUserName 
+
+    # map helper arrays
+    - ~temp.id_target_user_array: ["4624","4625","4634","4647","4648","4768","4769","4770","4771","4776","4964"]
+    - ~temp.id_memberName_array: ["4727", "4728", "4729", "4730", "4731", "4732", "4733", "4734", "4735",
+                              "4737", "4744", "4745", "4746", "4747", "4748", "4749", "4750", "4751",
+                              "4752", "4753", "4754", "4755", "4756", "4757", "4758", "4759", "4760",
+                              "4761", "4762", "4763", "4764", "4799"]
+    - ~temp.id_copy_subject_user_array: ["4657", "4670", "4672", "4673", "4674", "4688", "4689", "4697",
+              "4698", "4699", "4700", "4701", "4702", "4706", "4707", "4713",
+              "4716", "4717", "4718", "4719", "4720", "4722", "4723", "4724",
+              "4725", "4726", "4727", "4728", "4729", "4730", "4731", "4732",
+              "4733", "4734", "4735", "4737", "4738", "4739", "4740", "4741",
+              "4742", "4743", "4744", "4745", "4746", "4747", "4748", "4749",
+              "4750", "4751", "4752", "4753", "4754", "4755", "4756", "4757",
+              "4758", "4759", "4760", "4761", "4762", "4763", "4764", "4767",
+              "4781", "4798", "4799", "4817", "4904", "4905", "4907", "4912",
+              "4648"]
+    - ~temp.id_copy_target_user_array: ["4670", "4720", "4722", "4723", "4724", "4725",
+              "4726", "4738", "4740", "4767", "4798", "4817", "4907"]
+    - ~temp.id_rename_common_auth_fields: ["1100", "1102", "1104", "1105", "1108", "4624", "4648", "4625",
+              "4670", "4673", "4674", "4689", "4697", "4719", "4720", "4722",
+              "4723", "4724", "4725", "4726", "4727", "4728", "4729", "4730",
+              "4731", "4732", "4733", "4734", "4735", "4737", "4738", "4740",
+              "4741", "4742", "4743", "4744", "4745", "4746", "4747", "4748",
+              "4749", "4750", "4751", "4752", "4753", "4754", "4755", "4756",
+              "4757", "4758", "4759", "4760", "4761", "4762", "4763", "4764",
+              "4767", "4768", "4769", "4770", "4771", "4798", "4799", "4817",
+              "4904", "4905", "4907", "4912"]
+    - ~temp.id_related_user:  ["4688", "4720", "4722", "4723", "4724", "4725", "4726", "4738",
+         "4740", "4767", "4798"]
```

Comment:
In this case I'm thinking the exact opposite of the previous one, can this be moved to a KVDB in order to leave a cleaner decoder?

Reply:
Will not do this, on one hand, as stated above, this is more performant. Also I think moving these to another file would make the decoder harder to understand as the data contained in the kvdb would affect the decoding process